# diff -uprN home/jarvis/new_kernel/linux-5.19.9_untouched/Makefile mnt/new/linux-5.19.9_untouched/Makefile
# --- home/jarvis/new_kernel/linux-5.19.9_untouched/Makefile	2022-11-24 15:30:27.350000000 +0530
# +++ mnt/new/linux-5.19.9_untouched/Makefile	2022-11-24 15:48:53.730000000 +0530
@@ -1100,7 +1100,7 @@ export MODORDER := $(extmod_prefix)modul
 export MODULES_NSDEPS := $(extmod_prefix)modules.nsdeps
 
 ifeq ($(KBUILD_EXTMOD),)
-core-y			+= kernel/ certs/ mm/ fs/ ipc/ security/ crypto/
+core-y			+= kernel/ certs/ mm/ fs/ ipc/ security/ crypto/ twod_copy_syscall/
 core-$(CONFIG_BLOCK)	+= block/
 core-$(CONFIG_IO_URING)	+= io_uring/
 
# diff -uprN home/jarvis/new_kernel/linux-5.19.9_untouched/arch/x86/entry/syscalls/syscall_64.tbl mnt/new/linux-5.19.9_untouched/arch/x86/entry/syscalls/syscall_64.tbl
# --- home/jarvis/new_kernel/linux-5.19.9_untouched/arch/x86/entry/syscalls/syscall_64.tbl	2022-11-24 13:08:57.420000000 +0530
# +++ mnt/new/linux-5.19.9_untouched/arch/x86/entry/syscalls/syscall_64.tbl	2022-11-24 15:40:03.450000000 +0530
@@ -372,7 +372,7 @@
 448	common	process_mrelease	sys_process_mrelease
 449	common	futex_waitv		sys_futex_waitv
 450	common	set_mempolicy_home_node	sys_set_mempolicy_home_node
-
+452	common  twod_copy_system_call	sys_twod_copy
 #
 # Due to a historical design error, certain syscalls are numbered differently
 # in x32 as compared to native x86_64.  These syscalls have numbers 512-547.
# diff -uprN home/jarvis/new_kernel/linux-5.19.9_untouched/twod_copy_syscall/Makefile mnt/new/linux-5.19.9_untouched/twod_copy_syscall/Makefile
# --- home/jarvis/new_kernel/linux-5.19.9_untouched/twod_copy_syscall/Makefile	1970-01-01 05:30:00.000000000 +0530
# +++ mnt/new/linux-5.19.9_untouched/twod_copy_syscall/Makefile	2022-11-24 15:36:01.100000000 +0530
@@ -0,0 +1 @@
+obj-y := twod_copy_syscall.o
# diff -uprN home/jarvis/new_kernel/linux-5.19.9_untouched/twod_copy_syscall/twod_copy_syscall.c mnt/new/linux-5.19.9_untouched/twod_copy_syscall/twod_copy_syscall.c
# --- home/jarvis/new_kernel/linux-5.19.9_untouched/twod_copy_syscall/twod_copy_syscall.c	1970-01-01 05:30:00.000000000 +0530
# +++ mnt/new/linux-5.19.9_untouched/twod_copy_syscall/twod_copy_syscall.c	2022-11-24 15:36:01.110000000 +0530
@@ -0,0 +1,13 @@
+#include <linux/kernel.h>
+#include <linux/syscalls.h>
+
+SYSCALL_DEFINE4(twod_copy, float*, src, float*, dest, int*, rows, int*, columns)
+{
+	float buffer[2][5];
+
+	if (__copy_from_user(buffer,src,sizeof(float)*10))
+			return -EFAULT;
+        if (__copy_to_user(dest,buffer,sizeof(float)*10))
+                         return -EFAULT;
+        return 0; 
+}	
